text/0227-separate-semantic-conventions.md

```diff
@@ -0,0 +1,155 @@
+
+- Boilerplate files, e.g. `README.md`, `LICENSE`, `CODEOWNERS`, `CONTRIBUTING.md`
+- `Makefile` that allows automatic generation of documentation from model.
+- `model/` The set of YAML files that exist in
+  `{specification}/semantic_conventions` today.
+- `docs/` A new directory that contains human readable documentation for how to
+  create instrumentation compliant with semantic conventions.
+  - `resource/` - Contents of `{specification}/resource/semantic_conventions`
+  - `trace/` - Contents of `{specification}/trace/semantic_conventions`
+  - `metrics/` - Contents of `{specification}/metrics/semantic_conventions`
+  - `logs/`- Contents of `{specification}/logs/semantic_conventions`
+  - `schemas/` - A new location for [Telemetry Schemas](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/schemas/README.md)
+    to live. This directory will be hosted at
+    `https://opentelemetry.io/semantic_conventions/schemas/`
+
+Existing semantic conventions in the specification would be marked as
+deprecated, with documentation denoting the move.
+
+Additionally, if the semantic conventions eventually move to domain-specific
+director structure (e.g. `docs/{domain}/README.md`, with trace, metrics, events
+in the same file), then this can be refactored in the new repository, preserving
+git history.
+
+There will also be the following exceptions in the specification:
+
+- Semantic convetions used to implement API/SDK details will be fully specified
+  and will not be allowed to change in the Semantic Convention directory.
+  - Error/Exception handling will remain in the specification.
+  - SDK configuration interaction w/ semantic convention will remain in the
+    specification. Specifically `service.name`.
+- The specification may elevate some semantic conventions as necessary for
+  compatibility requirements, e.g. `service.instance.id` and
+  [Prometheus Compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md).
+
+These exceptions exist because:
+
+- Stable portions of the specification already rely on these conventions
+- These conventions are required to implement an SDK today.
+
+As such, the Specification should define the absolute minimum of reserved or
+required attribute names and their interaction with the SDK.
+
+## Internal details
+
+The following process would be used to ensure semantic conventions are
+seamlessly moved to their new location.
+
+- A new repository `opentelemetry/semantic_conventions` will be constructed with
+  the proposed format and necessary `Makefile` / tooling.
+- A moratorium will be placed on Semantic Convention PRs to the specififcation
+  repository. (Caveat that PRs related to this proposal would be allowed).
+- All semantic conventions in the Specification will be copied to the new
+  repository.
+- Semantic conventions in the Specification will be marked as deprecated with
```

Comment:
I suggest that we remove the semantic convention docs from the spec repo (for short-term, maybe just delete the contents and put a banner saying "things moved, here is the new location"), mainly due to the wording "deprecated" could introduce confusion.

Reply:
Maybe we do that eventually, but we need to deal with two realities:

1. We have in-flight data against the current SchemaURL of this repo.
2. There will be a transition period.

Reply:
> 1. We have in-flight data against the current SchemaURL of this repo.

I assume permanent links are used?

> 2. There will be a transition period.

100%

Reply:
Yes, we should use permanent links.

Reply:
My main concern here is the tie of SchemaURL to semantic-versioning specification and the fact we're already at `1.x`.

Given discussion around what is considered breaking changes and the role of SchemaURL here, I don't think we have had any guarantee of stability for Semconv yet.

I'd like to get off on the right foot here with a `0.x` release denoting instability and a `1.x` denoting stability.

Reply:
We discussed this and, like it or not, we have "defacto stable" semconv today from remaining unchanged for so long.  As such, starting at 1.x seems reasonable.  We're going to have the first release from the new repo be 2.0 as there are some breaking changes, and we can figure out version denotations going forward from there.